tt_metal/impl/dispatch/kernels/command_queue_consumer.hpp

```diff
@@ -288,6 +298,20 @@ FORCE_INLINE void wait_for_program_completion(uint32_t num_workers) {
     DEBUG_STATUS('Q', 'D');
 }
 
+// Record last completed event ID in dedicated L1 space for future event syncs to rely on.
+FORCE_INLINE void record_last_completed_event(uint32_t event_id) {
+    volatile tt_l1_ptr uint32_t* last_event_id = get_cq_completion_last_event();
+    last_event_id[0] = event_id;
+}
+// Cross CQ sync by waiting for a given CQ to have completed up to a certain event id.
+FORCE_INLINE void wait_for_event(uint32_t event_id, uint32_t noc_x, uint32_t noc_y) {
+    uint64_t src_noc_addr = get_noc_addr(noc_x, noc_y, CQ_COMPLETION_LAST_EVENT);
+    do {
+        noc_async_read(src_noc_addr, CQ_COMPLETION_16B_SCRATCH, 4);
```

Comment:
the `_SCRATCH` suffix is a little misleading because it isn't really being used as scratch space, it seems to be a sync location so maybe this can be renamed to `CQ_COMPLETION_SYNC`?

Reply:
We talked offline, I'm treating it as a scratch region that could be used as destination for any type of 16B noc read to L1 by this dispatcher core, but just using it here for the first time to read back the last-completed-event-id.